openadmet/models/features/mtenn.py

```diff
@@ -266,14 +270,22 @@ def featurize(self, complexes: Iterable[Path], y: Iterable[Any]) -> DataLoader:
         # if a pandas dataframe or series
         if isinstance(y, pd.DataFrame) or isinstance(y, pd.Series):
             y = y.to_numpy()
-        y = y.reshape(-1, 1)
 
-        self._dataset = MTENNDataset(
-            complexes=complexes,
-            y=y,
-            ligand_resname=self.ligand_resname,
-            ignore_h=self.ignore_h,
-        )
+        if y is not None:
+            y = y.reshape(-1, 1)
+
+            self._dataset = MTENNDataset(
+                complexes=complexes,
+                y=y,
+                ligand_resname=self.ligand_resname,
+                ignore_h=self.ignore_h,
+            )
+        else:
+            self._dataset = MTENNDataset(
+                complexes=complexes,
+                ligand_resname=self.ligand_resname,
+                ignore_h=self.ignore_h,
+            )
```

Comment:
Because Y is optional and can be none can you not just have a single branch? 

```python
            self._dataset = MTENNDataset(
                complexes=complexes,
                ligand_resname=self.ligand_resname,
                ignore_h=self.ignore_h,
                y=y # None is ok
            )
```